pkg/controllers/resourceapply/resourceapply_test.go

```diff
@@ -4,18 +4,84 @@ import (
 	"context"
 	"testing"
 
+	. "github.com/onsi/gomega"
+
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/utils/pointer"
-
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	appsclientv1 "sigs.k8s.io/controller-runtime/pkg/client"
-	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
+func setupEnvtest(t *testing.T) (client.Client, func(t *testing.T)) {
+	t.Log("Setup envtest")
+	g := NewWithT(t)
+	testEnv := &envtest.Environment{}
+	cfg, err := testEnv.Start()
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(cfg).NotTo(BeNil())
+
+	cl, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(cl).NotTo(BeNil())
+
+	teardownFunc := func(t *testing.T) {
+		t.Log("Stop envtest")
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}
+	return cl, teardownFunc
+}
+
+func cleanupResources(t *testing.T, g *WithT, ctx context.Context, cl client.Client, listObject client.ObjectList) {
+	g.Expect(cl.List(ctx, listObject)).To(Succeed())
+	deleteResouce := func(g *WithT, obj client.Object) {
+		key := client.ObjectKeyFromObject(obj)
+		obj.SetFinalizers([]string{})
+		g.Expect(cl.Update(ctx, obj)).To(Succeed())
+		err := cl.Delete(ctx, obj)
```

Comment:
Would be better to do this asynchronously in case the first time you do it it fails, you could retry

Reply:
Honestly i can not imagine why it might fail there.
I would prefer to let it crash instead of retry.